packages/blockchain-wallet-v4-frontend/src/modals/Swap/components/FeeBreakdownBox.tsx

```diff
@@ -0,0 +1,295 @@
+`
+const Footer = styled.div`
+  padding: 16px;
+`
+const FeeBreakdownBox = ({
+  base,
+  basePayment,
+  baseRates,
+  coins,
+  counter,
+  counterQuote,
+  counterRates,
+  paymentR,
+  quoteR,
+  supportedCoins,
+  walletCurrency
+}: Props): React.ReactElement => {
+  const [toggle, setToggle] = useState(false)
+  const networkFee = (value: PaymentValue | undefined) => {
+    return value
+      ? value.coin === 'BTC' || value.coin === 'BCH'
+        ? value.selection?.fee
+        : value.fee
+      : 0
+  }
+  const baseFiatFee = convertCoinToFiat(
+    (convertBaseToStandard(base.coin, networkFee(paymentR)) as unknown) as string,
+    base.coin as UnitType,
+    walletCurrency,
+    baseRates
+  )
+  const counterFiatFee = convertCoinToFiat(
+    (convertBaseToStandard(counter.coin, quoteR?.quote?.networkFee || '0') as unknown) as string,
+    counter.coin as UnitType,
+    walletCurrency,
+    counterRates
+  )
+  const counterName = supportedCoins[counter.coin]?.displayName || counter.coin
+  const baseName = supportedCoins[base.coin]?.displayName || base.coin
+  const bothCustodial = base.type === 'CUSTODIAL' && counter.type === 'CUSTODIAL'
+  const bothNonCustodial = base.type === 'ACCOUNT' && counter.type === 'ACCOUNT'
+
+  if (bothCustodial) {
+    return (
+      <Container>
+        <FeesContainer>
+          <HorizontalRow>
+            <Title>
+              <FormattedMessage id='copy.network_fees' defaultMessage='Network Fees' />
+            </Title>
+            <Value>
+              <FormattedMessage id='modals.brokerage.free' defaultMessage='Free' />
+            </Value>
+          </HorizontalRow>
+        </FeesContainer>
+        <Footer>
+          <Text size='12px'>
+            <FormattedMessage
+              id='copy.no_network_fees'
+              defaultMessage='When using Trading Accounts, there are no network fees.'
+            />
+          </Text>
+        </Footer>
+      </Container>
+    )
+  }
+
+  return (
+    <Container>
+      <FeesContainer>
+        <HorizontalRow>
+          <Title>
+            <FormattedMessage id='copy.network_fees' defaultMessage='Network Fees' />
+          </Title>
+          <Value>
+            <HorizontalRow style={{ border: 'unset', padding: '0px' }}>
+              <Text color='grey900'>~</Text>
+              <FiatDisplay size='14px' weight={500} color='grey900' coin={walletCurrency}>
+                {Number(baseFiatFee) + Number(counterFiatFee)}
+              </FiatDisplay>
+              <Icon
+                role='button'
+                data-e2e='toggleSwapFeesDropdown'
+                name='chevron-down'
+                cursor
+                size='24px'
+                color='grey900'
+                onClick={() => {
+                  setToggle((prev) => !prev)
+                }}
+              />
+            </HorizontalRow>
+          </Value>
+        </HorizontalRow>
+        {toggle && (
+          <>
+            {base.type === 'ACCOUNT' && (
```

Comment:
can/should we use the `address_types` enum? e.g.`ADDRESS_TYPES.ACCOUNT`

Reply:
Yep!